src/lib/storage.js

```diff
@@ -360,28 +360,33 @@ class Storage {
       uplink with proxy_access rights against {name} and combines results
       into one json object
     Used storages: local && uplink (proxy_access)
-   * @param {*} name
-   * @param {*} options
-   * @param {*} callback
-   */
-  get_package(name, options, callback) {
-    if (_.isFunction(options)) {
-      callback = options, options = {};
-    }
 
-    this.localStorage.getPackageMetadata(name, (err, data) => {
+   * @param {object} options
+   * @property {string} options.name Package Name
+   * @property {object}  options.req Express `req` object
+   * @property {boolean} options.keepUpLinkData keep up link info in package meta, last update, etc.
+   * @property {function} options.callback Callback for receive data
+   */
+  get_package(options) {
+    this.localStorage.getPackageMetadata(options.name, (err, data) => {
       if (err && (!err.status || err.status >= 500)) {
         // report internal errors right away
-        return callback(err);
+        return options.callback(err);
       }
 
-      this._syncUplinksMetadata(name, data, options, function(err, result, uplink_errors) {
+      this._syncUplinksMetadata(options.name, data, {req: options.req}, function(err, result, uplink_errors) {
         if (err) {
-          return callback(err);
+          return options.callback(err);
+        }
+
+        const propertyToKeep = [];
+        Array.prototype.push.apply(propertyToKeep, WHITELIST);
+        if (options.keepUpLinkData === true) {
```

Comment:
Might it be something else?  In case to skip `undefined` you might use `_.isNil`

Reply:
Make sure it's strict, so, will not change it